src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -98,13 +85,36 @@ def update_sold_bought_energy(self, area: Area):
                     self.traded_energy_current, area.uuid)
                 self.time_slots = generate_market_slot_list(area)
 
-            self.balancing_traded_energy[area.uuid] = self.balancing_traded_energy[area.name]
+                # Merges traded energy for the CSV file
+                # TODO: Adapt to not store the full results on D3A.
+                if area.slug not in self.traded_energy_profile:
+                    self.traded_energy_profile[area.slug] = \
+                        {"sold_energy": {}, "bought_energy": {}}
+                    self._calculate_devices_sold_bought_energy(
+                        self.traded_energy_profile[area.slug], area.current_market)
+                    self.traded_energy_profile[area.slug] = self._serialize_traded_energy_lists(
+                        self.traded_energy_profile, area.slug)
+                else:
+                    traded_energy_current_name = {area.slug: self.traded_energy_current[area.uuid]}
+                    self.traded_energy_profile = merge_energy_trade_profile_to_global(
+                        traded_energy_current_name, self.traded_energy_profile,
+                        generate_market_slot_list(area))
+
+            if self._should_export_plots:
+                # Traded energy for plot
+                if area.uuid not in self.traded_energy:
+                    self.traded_energy[area.uuid] = {"sold_energy": {}, "bought_energy": {}}
+                self._calculate_devices_sold_bought_energy(self.traded_energy[area.uuid],
+                                                           area.current_market)
+                self.traded_energy[area.name] = self.traded_energy[area.uuid]
+                if area.name not in self.balancing_traded_energy:
+                    self.balancing_traded_energy[area.name] = {"sold_energy": {},
+                                                               "bought_energy": {}}
+                if len(area.past_balancing_markets) > 0:
+                    self._calculate_devices_sold_bought_energy(
+                        self.balancing_traded_energy[area.name], area.current_balancing_market)
```

Comment:
I think this whole mechanism could be refactored and simplified in the frame of 2212.
But thanks for the starting point!

Reply:
I agree, it is definitely worth a refactoring. Adding it in the description of 2212.

Reply:
Thanks!